### PR TITLE
scrapers now control starting and saving assignments

### PR DESCRIPTION
**Pull request by calebjclark**
- AssignmentServer now controls it's own logic (I pulled route logic from DetectionsServer and instead just reference the vars needed).
- AssignmentServer no longer controls the order of assignments. The scraper fetches its list (`/`) then calls start (`/start`) on each one. It then calls finish (`/finish`) to retrieve results for saving.
- Only session/ files are saved: botStats.json, bucketStats.json and browser-flags/ are no longer created.
- Most of this was done in the process of me trying to figure out the flow... it also now separates the running from the processing of results, making it easy to move scrapers into `scraper-report`.

### CODE REVIEW

File: runner/detections-server/httpRequestHandler.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export default function httpRequestHandler(
       return res.end();
     }
 
+    console.log('Detections %s', `${req.headers.host}${req.url}`);
```
[the remark this answers is not in the document]
Comment (calebjclark) [reply runner/detections-server/httpRequestHandler.ts:29]: Line 67 doesn't always run if there are issues. What do you think about cutting `botScore` from line 67 and moving it up to replace this one?

File: scrapers/lib/BotDetectionResults.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export default class BotDetectionResults {
         const entry = browserFinding[flaggedCheck.category];
         entry.flagged += 1;
         entry.botPct = Math.max(entry.botPct ?? 0, flaggedCheck.pctBot);
-        entry.flaggedChecks.push(flaggedCheck);
+        // entry.flaggedChecks.push(flaggedCheck);
```
[the remark this answers is not in the document]
Comment (calebjclark) [reply scrapers/lib/BotDetectionResults.ts:37]: Yes. For now. 

The flaggedChecks were being deleted before printing to console.log. It was only saved to the botStats.json. Now that file is gone. flaggedChecks is way too verbose for stdout. However, I didn't want to delete this... might be good for debugging.

File: runner/detections-server/httpRequestHandler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (blakebyrnes) [reply runner/detections-server/httpRequestHandler.ts:29]: Botscore is something I've used pretty often when I try to update secret agent fixes to see if they're working.

[the remark this answers is not in the document]
Comment (blakebyrnes) [reply runner/detections-server/httpRequestHandler.ts:29]: Won't it log the catch log if it doesn't get hit? Are you sure this is catching a missed condition?

[the remark this answers is not in the document]
Comment (calebjclark) [reply runner/detections-server/httpRequestHandler.ts:29]: Fair.

Most servers log requests at the start so you know the server was hit and with what route. Not logging until everything has run is a bit odd. I wonder if botScore should be displayed on line 67, but the rest (req) should be displayed earlier. Thoughts?

**Pull request merged**